release-lock-file-spec.md

```diff
@@ -0,0 +1,188 @@
+
+### Version: `version`
+
+The `version` field declares the version number of this release.  This value
+**must** be included in all release lock files.  This value **should** be conform
+to the [semver](http://semver.org/) version numbering specification.
+
+* Key: `version`
+* Type: String
+* Format: [semver](http://semver.org)
+
+
+### Sources: `sources`
+
+The `sources` field defines a source tree that **should** comprise the full
+source tree necessary to recompile the contracts contained in this release.
+Sources are declared in a key/value mapping.  
+
+* All keys **must** conform to *one of* the following formats.
+    * Begins with a `./` to denote that it is a filesystem path.
+    * Is a valid contract name matching the regex `[_a-zA-Z][_a-zA-Z0-9]*`.
+* All keys which are formatted to be filesystem paths **must** conform to *all* of the following rules:
+    * Resolving the filesystem paths must result path that is *under* the current working directory.
+
+* *If* the key is a contract name the value **must** be the source string for that contract.
+* *If* the key is a filesystem path the value **must** conform to *one of* the following formats.
+    * IPFS URI
+        * *If* the resulting document is a directory the key should be interpreted as a directory path.
+        * *If* the resulting document is a file the key should be interpreted as a file path.
+
+* Key: `sources`
+* Type: Hash(String: String)
+
+
+### Chain: `chain`
+
+The `chain` field declares the blockchain that the contract addresses apply to.
+Chains are defined by as a key/value mapping with block numbers as keys and
+block hashes as values.  Both block numbers and hashes **must** be hexidecimal
+encoded.  Convention is to define a chain using two blocks, the genesis block
+under the key `0x00` and the latest observable block at the time of release.
+*If* this release lock file includes any addressed contracts this field
+**must** be present.
+
+* Key: `chain`
+* Type: Hash(BlockNumber: BlockHash)
+
+
+### Contracts: `contracts`
+
+The `contracts` field declares information about the deployed contracts.
+
+* a single *Contract Instance Object*
+* a List of *Contract Instance Objects*
+
+* Key: `contracts`
+* Type:  *any of:*
+    - *Contract Instance Object* 
+    - List of *Contract Instance Object* 
+    - Hash of (String: *Contract Instance Object*) where keys are valid contract names matching the regex `[_a-zA-Z][_a-zA-Z0-9]*`.
+
+
+#### The *Contract Instance Object*
+
+A *Contract Instance Object* is an object with the following key/values.
+
+* `contract_name`:
+    * Required: Yes
+    * Type: String
+    * Format: Valid contract name matching regular expression `[_a-zA-Z][_a-zA-Z0-9]*]`
+* `address`:
+    * Required: No
+    * Type: String
+    * Format: Hex encoded ethereum address of the deployed contract.
+* `bytecode`:
+    * Required: No
+    * Type: String
+    * Format: Hex encoded unlinked bytecode for the compiled contract.
+* `runtime_bytecode`:
```

Comment:
What is the difference between runtime_bytecode and normal bytecode? I also strongly believe that at the very least the ABI needs to be required. Quite possibly the bytecode too but not sure about that.